druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/MahaRegisteredLookupExtractionFn.java

```diff
@@ -60,6 +69,7 @@ public MahaRegisteredLookupExtractionFn(
         this.valueColumn = valueColumn;
         this.decodeConfig = decodeConfig;
         this.dimensionOverrideMap = dimensionOverrideMap;
+        this.useQueryLevelCache = useQueryLevelCache == null ? false : useQueryLevelCache;
```

Comment:
Why not instantiate cache here if the above is set to true?

Reply:
I need to lazily instantiate cache. If I eagerly instantiate then maha-core which uses maha-druid-lookups while generating druid query unnecessarily instantiates cache.